cirq-core/cirq/transformers/transformer_primitives_test.py

```diff
@@ -205,6 +205,35 @@ def map_func(op: cirq.Operation, _: int) -> cirq.OP_TREE:
 # pylint: enable=line-too-long
 
 
+@pytest.mark.parametrize("deep", [False, True])
+def test_map_operations_preserves_circuit_tags(deep: bool) -> None:
+    tag = "should be preserved"
+
+    def func(op: cirq.Operation, idx: int) -> cirq.Operation:
+        return cirq.Y(op.qubits[0]) if op.gate == cirq.X else op
+
+    x = cirq.X(cirq.q(0))
+    circuit = cirq.FrozenCircuit.from_moments(x, cirq.FrozenCircuit(x)).with_tags(tag)
+    mapped = cirq.map_operations(circuit, func, deep=deep)
+
+    assert circuit.tags == (tag,)
```

Comment:
why do we check the input circuit here and below?

Reply:
They're not needed. I will remove because I agree the point is not clear.